src/internal/methods/NeoEsp32LcdXMethod.h

```diff
@@ -0,0 +1,551 @@
+        // i2sDeinit(busNumber);
+
+        gdma_reset(dma_chan);
+
+        heap_caps_free(LcdBuffer);
+
+        LcdBufferSize = 0;
+        LcdBuffer = nullptr;
+
+        MuxMap.Reset();
+    }
+
+    void ResetBuffer()
+    {
+        // to keep the inner loops for EncodeIntoDma smaller
+        // they will just OR in their values
+        // so the buffer must be cleared first
+        if (MuxMap.IsNoMuxBusesUpdate())
+        {
+            // clear all the data in preperation for each mux channel to add
+            memset(LcdBuffer, 0x00, LcdBufferSize);
+        }
+    }
+
+    void FillBuffer(uint8_t** dmaBuffer,
+            const uint8_t* data, 
+            size_t sizeData, 
+            uint8_t muxId)
+    {
+        MuxMap.EncodeIntoDma(dmaBuffer,
+            data,
+            sizeData,
+            muxId);
+    }
+
+    void StartWrite()
+    {
+        if (MuxMap.IsAllMuxBusesUpdated())
+        {
+            MuxMap.ResetMuxBusesUpdated();
+            
+            gdma_reset(dma_chan);
+            LCD_CAM.lcd_user.lcd_dout = 1;
+            LCD_CAM.lcd_user.lcd_update = 1;
+            LCD_CAM.lcd_misc.lcd_afifo_reset = 1;
+
+            uint32_t xfer_size = MuxMap.MaxBusDataSize * DmaBytesPerPixelByte;
+            int num_desc = (xfer_size + 4094) / 4095; // sic. (NOT 4096)
+            int bytesToGo = xfer_size;
+            int offset = 0;
+            for (int i = 0; i < num_desc; i++) {
+                int bytesThisPass = bytesToGo;
+                if (bytesThisPass > 4095)
+                bytesThisPass = 4095;
+                desc[i].dw0.size = desc[i].dw0.length = bytesThisPass;
+                desc[i].buffer = &DmaBuffer[offset];
+                bytesToGo -= bytesThisPass;
+                offset += bytesThisPass;
+            }
+
+            gdma_start(dma_chan, (intptr_t)&desc[0]);
+            esp_rom_delay_us(1);
+            LCD_CAM.lcd_user.lcd_start = 1; 
+        }
+    }
+};
+
+//
+// Implementation of the low level interface into i2s mux bus
+// 
+// T_BUSCONTEXT - the context to use, currently only NeoEspLcdMonoBuffContext
+//
+template<typename T_BUSCONTEXT> 
+class NeoEsp32LcdMuxBus
+{
+public:    
+    NeoEsp32LcdMuxBus() :
+        _muxId(s_context.MuxMap.InvalidMuxId)
+    {
+    }
+
+    void RegisterNewMuxBus(size_t dataSize)
+    {
+        _muxId = s_context.MuxMap.RegisterNewMuxBus(dataSize);
+    }
+
+    void Initialize(uint8_t pin)
+    {
+        s_context.Construct();
+
+        uint8_t muxIdx = LCD_DATA_OUT0_IDX + _muxId;
+        esp_rom_gpio_connect_out_signal(pin, muxIdx, false, false);
+        gpio_hal_iomux_func_sel(GPIO_PIN_MUX_REG[pin], PIN_FUNC_GPIO);
+        gpio_set_drive_capability((gpio_num_t)pin, (gpio_drive_cap_t)3);
+    }
+
+    void DeregisterMuxBus(uint8_t pin)
+    {
+        if (s_context.MuxMap.DeregisterMuxBus(_muxId))
+        {
+            s_context.Destruct();
+        }
+
+        gpio_matrix_out(pin, SIG_GPIO_OUT_IDX, false, false);
```

Comment:
they may have a new detach instead of calling this, but this is fine.
Really only commenting about this as a note to a previous comment about the similar detaching, as this is the correct place to do this.